app/controllers/carts_controller.rb

```diff
@@ -1,17 +1,21 @@
 class CartsController < ApplicationController
+  before_action :not_shop_account?
```

Comment:
Insufficient check? This should have checked for normal user sign in status.
```ruby
before_action :user_signed_in?
```

Reply:
luc add cart thi ko can dang nhap user cung duoc em, nhung dang nhap bang shop_account thi khong cho mua hang.